### ISSUE TITLE
[Question] How the dependency feature suppose to work?

### ISSUE THREAD

**Issue by ricardozanini** [operator-framework/operator-lifecycle-manager#1010]
I was looking for a feature to add two Operators as dependencies to a custom operator that I'm working on. I understand that the OLM dependency feature would perfectly fit for my use case.

After reading the docs ([Dependency Resolution](https://github.com/operator-framework/operator-lifecycle-manager#dependency-resolution)) and ([Required CRDs](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/Documentation/design/building-your-csv.md#required-crds)), I updated my `ClusterServiceVersion` to add the required dependencies as follows:

```yaml
customresourcedefinitions:
    required:
      - name: infinispans.infinispan.org
        version: v1
        kind: Infinispan
        displayName: Infinispan Cluster
        description: Represents a Infinispan cluster used internally by the Data Index Service
      - name: kafkas.kafka.strimzi.io
        version: v1beta1
        kind: Kafka
        displayName: Strimzi Kafka Cluster
        description: Represents a Kafka cluster used internally by the Data Index Service
    owned:
      - version: v1alpha1
        kind: MyKind
        description: My Kind Description
        displayName: MyKindDisplayName
        name: mykind.namespace.example.org
```

In this case, this operator will leverage from Infinispan and Kafka operators.

I deployed the application into Quay and pushed using `operator-courier`. Then created an entry into the OperatorHub:

```yaml
apiVersion: operators.coreos.com/v1
kind: OperatorSource
metadata:
  name: dummy-operator
  namespace: openshift-marketplace
spec:
  type: appregistry
  endpoint: https://quay.io/cnr
  registryNamespace: ricardozanini
  displayName: "My Operator"
```

After that, I created a subscription to the operator into a new namespace. The operator wasn't installed, although the `installed-custom-namespace-5767c568db-85ghx` pod and the subscription were successfully created.

What features I was expecting:

1. During the subscription, an interface hint about the required dependencies for this custom operator taken from the `ClusterServiceVersion` file (`required.displayName` and `required.description`)
2. Those dependencies automatically installed together with my custom operator

After manually installing both dependencies, my custom operator was successfully deployed. This is nice since it's enforcing the dependencies, but from an user point of view I missed the mentioned expected features. 

I suppose I did something wrong. I was hoping that you could help me figure this out. :smiley: 

Many thanks!

**Comment by ecordell** [comment]
Hi @ricardozanini  - Your expectations are spot on. 

Am I guessing correctly that you are using openshift 4.1.z? If so, there is an issue in 4.1.z that prevents OLM from doing automatic dependency resolution. 

The easiest workaround is, as you suggest, to manually select the dependency to install. This issue is fixed in 4.2.

**Comment by ricardozanini** [comment]
Hi @ecordell!

Yes, I'm on 4.1.z, that's the reason why I'm seeing this then. Many thanks for your explanation. Makes sense, I'll set the expectations of my team regarding this issue on 4.1.z.
